### ISSUE TITLE
Caret is visible when it leaves CodeBlock

### ISSUE THREAD

**Issue by willurmston** [remirror/remirror#961]
### Summary

When you put the caret at the end of a CodeBlock and press the Right arrow key, the GapCursor appears below the block. The normal text caret is still visible, leading users to think that text will be entered in Code Block. As an aside, I was surprised to see the GapCursor at all, because I initially tried to fix this issue by excluding it from the `corePreset` with:
```
const extensions = corePreset({ excludeExtensions: ['gapCursor'] });
```

### Steps to reproduce

Remirror version: 0.0.0-pr706.23
CodeSandbox: https://codesandbox.io/s/remirror-pr-000-pr70623-forked-5gfvr?file=/src/App.tsx

1. Create a Code Block
2. Enter some text
3. Press the Right Arrow key

### Expected results

The standard caret should not be visible.

### Actual results

The caret is visible.

### Screenshot(s)

https://user-images.githubusercontent.com/8092948/122126394-83107c80-cde6-11eb-88c7-a8d78afdab3b.mp4

**Comment by ifiokjr** [comment]
Thanks for this well-structured issue.

Try this link - https://codesandbox.io/s/remirror-pr-000-pr70623-forked-purbm

The changes made are to add the `TrailingNodeExtension` which makes it possible to escape the Code Block without using the <kbd>Cmd-Enter</kbd> or <kbd>Ctrl-Enter</kbd> shortcuts. 

You can pass the settings for the `CorePreset` to the `useRemirror` hook.

**Comment by willurmston** [comment]
@ifiokjr Thanks for your help! Unfortunately in our use-case we can't have empty nodes hanging out at the end the text block because we align it to other elements. 

I am actually able to escape the codeblock using the ➡️ / ⬇️ keys in the sandbox I linked to. Is that not by design? To clarify, I like the GapCursor shows up in that situation, I just wish the _text_ caret disappeared to avoid confusion.

**Comment by ifiokjr** [comment]
@willurmston I think I understand now. 

The issue is probably something that would need to be fixed in `prosemirror-gapcursor`. Since the `codeBlock` is the last text block, pressing the ➡️/⬇️ keys does not move the (vertical) cursor. I'm not sure how to remove the vertical cursor, since that's built in to the browser.  

The problem is that the `gapcursor` activates with its own (horizontal) cursor. 

A workaround would be to create an arrow key listener that checks

- Is the cursor at the end of a `codeBlock`?
- Is the `codeBlock` at the end of the document?
- If YES to both, create a new default block node and step the cursor into it. 

The `gapcursor` won't appear, and you won't have the double flashing cursors.

I may eventually add a utility function to `prosemirror-trailing-node` which makes implementing the solution easier, but it's not a priority and won't be done for a while. 

I'm open to a PR if you have the time.

**Comment by willurmston** [comment]
Thanks, that sounds like a good workaround. Alternatively, after some cursory (sorry) research it seems like we could also hide the vertical caret with `caret-color: transparent;` when the `gapcursor` is visible. I'm not sure where that CSS override should happen though. I may have some time to look into it after our impending release.

**Comment by ifiokjr** [comment]
Turns out this is already built into `prosemirror-gapcursor`. I had just set up the styles wrong 🤦 

I've pushed up a fix in #973 and will release a `pr706` once the CI checks are completed.
